### DISCUSSION TITLE
ETH Not deducted when made a contract

### DISCUSSION THREAD

**Discussion by MahzyarSaadat** [ConsenSys-archive/ganache#4415] [Q&A]
I have created a contract for the first time by ethers. Although my code worked, my Ganache account has not deducted any money. (It shows 4 transaction because I tried it multiple times);
![239946796-9f9bd3df-7817-4bb6-b86b-bb5222e6beb7](https://github.com/trufflesuite/ganache/assets/108320191/d7dda08f-1295-44c4-af25-12a92dad9255)
Here is my code:
```javascript
const ethers = require('ethers');
const fs = require('fs');

async function main() {
    const provider = new ethers.JsonRpcProvider("http://127.0.0.1:7545");
    const wallet = new ethers.Wallet("0x81ca952f26d9b1f59a5a5c88b87d6dcff44af4ed2869d98b567053310fb116b2", provider);

    const bin = fs.readFileSync("./first_sol_SimpleStorage.bin", "utf-8");
    const abi = fs.readFileSync("./first_sol_SimpleStorage.abi", "utf-8");

    const contractFactory = new ethers.ContractFactory(abi, bin, wallet);
    console.log("Deploying, Please wait!!");
    const contract = await contractFactory.deploy();
    console.log(contract);

    // contract.deploy()
}

main()
    .then(() => console.log("Success"))
    .catch(error => console.log("Some error happened => ", error))
```
My output:
![239947379-49061ec9-ddd3-4aaf-80ab-671478d286be](https://github.com/trufflesuite/ganache/assets/108320191/ef63371a-37a6-4429-ba1b-0e511e0164da)
Ganach V: v2.7.1
ethers V: v6.4.0
node V: v18.12.1

**Comment by davidmurdoch** [comment]
According to https://docs.ethers.org/v5/api/contract/contract-factory/#ContractFactory-deploy I think you need to do:

```
const contract = await contractFactory.deploy();
console.log(contract);
    
await contract.deployTransaction.wait();

```
